packages/commonwealth/client/scripts/state/api/threads/fetchThreads.ts

```diff
@@ -0,0 +1,207 @@
+  queryType: typeof QueryTypes.ACTIVE; // discriminating union
+  topicsPerThread?: number;
+}
+
+const featuredFilterQueryMap = {
+  newest: 'createdAt:desc',
+  oldest: 'createdAt:asc',
+  mostLikes: 'numberOfLikes:desc',
+  mostComments: 'numberOfComments:desc',
+};
+
+const useDateCursor = ({
+  dateRange,
+}: {
+  dateRange?: ThreadTimelineFilterTypes;
+}) => {
+  const [dateCursor, setDateCursor] = useState<{
+    toDate: string;
+    fromDate: string | null;
+  }>({ toDate: moment().toISOString(), fromDate: null });
+
+  useEffect(() => {
+    const updater = () => {
+      const { toDate, fromDate } = getToAndFromDatesRangesForThreadsTimelines(dateRange)
+      setDateCursor({ toDate, fromDate });
+    };
+
+    // set date cursor and schedule it to run after some intervals
+    updater();
+    const interval = setInterval(() => updater(), THREADS_STALE_TIME - 10);
+
+    return () => {
+      clearInterval(interval);
+    };
+  }, [dateRange]);
+
+  return { dateCursor };
+};
+
+const isFetchActiveThreadsProps = (props): props is FetchActiveThreadsProps =>
+  props.queryType === QueryTypes.ACTIVE;
+
+const isFetchBulkThreadsProps = (props): props is FetchBulkThreadsProps =>
+  props.queryType === QueryTypes.BULK;
+
+const getFetchThreadsQueryKey = (props) => {
+  if (isFetchBulkThreadsProps(props)) {
+    return [
+      ApiEndpoints.FETCH_THREADS,
+      props.chainId,
+      props.queryType,
+      props.topicId,
+      props.stage,
+      props.includePinnedThreads,
+      props.toDate,
+      props.fromDate,
+      props.limit,
+      props.orderBy,
+    ];
+  }
+  if (isFetchActiveThreadsProps(props)) {
+    return [
+      ApiEndpoints.FETCH_THREADS,
+      props.chainId,
+      props.queryType,
+      props.topicsPerThread,
+    ];
+  }
+}
+
+const fetchBulkThreads = (props) => {
+  return async ({ pageParam = 1 }) => {
+    const res = await axios.get(
+      `${app.serverUrl()}${ApiEndpoints.FETCH_THREADS}`,
+      {
+        params: {
+          bulk: true,
+          page: pageParam,
+          limit: props.limit,
+          chain: props.chainId,
+          ...(props.topicId && { topic_id: props.topicId }),
+          ...(props.stage && { stage: props.stage }),
+          ...(props.includePinnedThreads && {
+            includePinnedThreads: props.includePinnedThreads || true,
+          }),
+          ...(props.fromDate && { from_date: props.fromDate }),
+          to_date: props.toDate,
+          orderBy:
+            featuredFilterQueryMap[props.orderBy] ||
+            featuredFilterQueryMap.newest,
+        },
+      }
+    );
+
+    // transform the response
+    const transformedData = {
+      ...res.data.result,
+      threads: res.data.result.threads.map((c) => new Thread(c)),
+    };
+
+    return {
+      data: transformedData,
+      pageParam:
+        transformedData.threads.length > 0 ? pageParam + 1 : undefined,
+    };
+  };
+}
+
+const fetchActiveThreads = (props) => {
```

Comment:
We should really come up with a plan for cleaning up these 3 queries, as you mention. Out of scope for this PR, but seems necessary.